src/main/java/org/jenkinsci/plugins/p4/workspace/ManualWorkspaceImpl.java

```diff
@@ -172,12 +172,34 @@ else if (client != null && adjustClientName) {
 	}
 
 
-	private ArrayList<String> getChangeView(WorkspaceSpec workspaceSpec) {
+	private ArrayList<String> getChangeView(IOptionsServer connection, WorkspaceSpec workspaceSpec) throws Exception
+    {
 		ArrayList<String> changeView = new ArrayList<>();
 
 		String view = workspaceSpec.getChangeView();
 		if (view != null) {
 			String specString = getExpand().format(view, true);
+			boolean fromFile = false;
+			if (specString.startsWith("@")) {
+				// extract View from file:	JENKINS-69491.
+				fromFile = true;
+				logger.fine("P4: view from file=" + specString);
+				String specPathFull = specString.substring(1).trim();
+				List<IFileSpec> file = FileSpecBuilder.makeFileSpecList(specPathFull);
+				GetFileContentsOptions printOpts = new GetFileContentsOptions();
+				printOpts.setNoHeaderLine(true);
+				InputStream ins = null;
+				try {
+					connection.getFileContents(file, printOpts);
```

Comment:
file content is never assigned, so ins will always be null.

Reply:
yes, line 194 should have been the following of course - guess I forgot to push that before creating the PR
ins = connection.getFileContents(file, printOpts);